src/curve/stable/StableMintBurn.sol

```diff
@@ -137,40 +135,39 @@ contract StableMintBurn is ReservoirPair {
 
         _burn(address(this), liquidity);
 
-        _checkedTransfer(token0, to, amount0, lReserve0, lReserve1);
-        _checkedTransfer(token1, to, amount1, lReserve0, lReserve1);
+        _checkedTransfer(token0, aTo, amount0, lReserve0, lReserve1);
+        _checkedTransfer(token1, aTo, amount1, lReserve0, lReserve1);
 
-        _update(_totalToken0(), _totalToken1(), uint104(lReserve0), uint104(lReserve1));
-
-        lastInvariant = uint192(_computeLiquidity(_slot0.reserve0, _slot0.reserve1));
+        uint256 lBalance0 = _totalToken0();
+        uint256 lBalance1 = _totalToken1();
+        lastInvariant = uint192(_computeLiquidity(lBalance0, lBalance1));
         lastInvariantAmp = _getCurrentAPrecise();
-
         emit Burn(msg.sender, amount0, amount1);
 
+        _updateAndUnlock(lBalance0, lBalance1, uint104(lReserve0), uint104(lReserve1), lBlockTimestampLast);
```

Comment:
Are these `uint104` casts extraneous/no-ops?

Reply:
Can merge after this is resolved

Reply:
yes they seem superfluous, and the compiler actually makes it a no-op. 

Fixed